cpp/src/arrow/flight/sql/example/sqlite_type_info.cc

```diff
@@ -0,0 +1,116 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "arrow/flight/sql/example/sqlite_type_info.h"
+
+#include "arrow/flight/sql/server.h"
+#include "arrow/flight/sql/types.h"
+#include "arrow/record_batch.h"
+#include "arrow/testing/gtest_util.h"
+
+namespace arrow {
+namespace flight {
+namespace sql {
+namespace example {
+
+std::shared_ptr<RecordBatch> DoGetTypeInfoResult() {
+  auto type_name_array =
+      ArrayFromJSON(utf8(), R"(["bit", "tinyint", "bigint", "longvarbinary",
+                            "varbinary", "text", "longvarchar", "char",
+                            "integer", "smallint", "float", "double",
+                            "numeric", "varchar", "date", "time",
+                            "timestamp"])");
+  auto data_type = ArrayFromJSON(
+      int32(), R"([-7, -6, -5, -4, -3, -1, -1, 1, 4, 5, 6, 8, 8, 12, 91, 92, 93])");
```

Comment:
Hmm, where do 91, 92 and 93 come from? They don't seem to correspond to any `XDBCDataType` enum value?

Reply:
Actually, judging by https://github.com/aosm/iodbc/blob/d08b69dda97da494821a34ffb8aa56b4bac50b0f/iodbc/include/sql.h#L181-L185, these are standard in ODBC >= 3.0.0 (which was released in 1995)? Perhaps we should add those values to the proto file as well?

Reply:
Yeah, it makes complete sense. Since they are in a different file in the odbc I've might missed it.

Added